CodePush.ios.js

```diff
@@ -95,12 +96,86 @@ function notifyApplicationReady() {
+ * 
+ * It simply composes the existing API methods together and adds additional
+ * support for respecting mandatory updates, ignoring previously failed
+ * releases, and displaying a standard confirmation UI to the end-user
+ * when an update is available.
+ */
+function sync(options = {}) {  
+  var syncOptions = {
+    ignoreFailedUpdates: true,
+    
+    mandatoryContinueButtonLabel: "Continue",
+    mandatoryUpdateMessage: "An update is available that must be installed",
+    
+    optionalIgnoreButtonLabel: "Ignore",
+    optionalInstallButtonLabel: "Install",
+    optionalUpdateMessage: "An update is available. Would you like to install it?",
+    
+    updateTitle: "Update available",
+    rollbackTimeout: 0,
+    
+    ...options 
+  };
+
+  return new Promise((resolve, reject) => {
+    checkForUpdate()
+      .then((remotePackage) => {
+        if (!remotePackage || (remotePackage.failedAppy && syncOptions.ignoreFailedUpdates)) {
+          resolve(CodePush.SyncStatus.NO_UPDATE_AVAILABLE);
+        }
+        else {
+          var message = null;
+          var dialogButtons = [
+            {
+              text: null,
+              onPress: () => { 
+                remotePackage.download()
+                  .then((localPackage) => {
+                    resolve(CodePush.SyncStatus.APPLY_SUCCESS);
+                    localPackage.apply(syncOptions.rollbackTimeout);
+                  }, reject);
+              }
+            }
+          ];
+          
+          if (remotePackage.isMandatory) {
+            message = syncOptions.mandatoryUpdateMessage;
+            dialogButtons[0].text = syncOptions.mandatoryContinueButtonLabel;
+          } else {
+            message = syncOptions.optionalUpdateMessage;
+            dialogButtons[0].text = syncOptions.optionalInstallButtonLabel;     
+            
+            // Since this is an optional update, add another button
+            // to allow the end-user to ignore it       
+            dialogButtons.push({
+              text: syncOptions.optionalIgnoreButtonLabel,
+              onPress: () => resolve(CodePush.SyncStatus.UPDATE_IGNORED)
+            });
+          }
+          
+          AlertIOS.alert(syncOptions.updateTitle, message || remotePackage.description, dialogButtons);
+        }
+      }, reject);
+  });     
+};
+
 var CodePush = {
   getConfiguration: getConfiguration,
   checkForUpdate: checkForUpdate,
   getCurrentPackage: getCurrentPackage,
   notifyApplicationReady: notifyApplicationReady,
-  setUpTestDependencies: setUpTestDependencies
+  setUpTestDependencies: setUpTestDependencies,
+  sync: sync,
+  SyncStatus: {
+    NO_UPDATE_AVAILABLE: 0, // The running app is up-to-date
+    UPDATE_IGNORED: 1, // The app had an optional update and the end-user chose to ignore it
+    APPLY_SUCCESS: 2 // The app had an optional/mandatory update that was successfully downloaded and is about to be applied
```

Comment:
The field name and comment describe different states.  APPLY_SUCCESS means apply succeeded; however, the comment says the "... update ... is about to be applied."  Given the implementation above the field name should be changed to DOWNLOAD_SUCCEEDED.